spec/helpers/jwt_auth_helpers_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+RSpec.describe JWTAuthHelpers, type: :helper do
+  describe '#user_info_from_header' do
+    it 'should return nil when the request does not have an Authorization header' do
+      allow(helper).to receive(:headers).and_return({})
+      expect(helper.send(:user_info_from_header)).to eq(nil)
```

Comment:
This is a religious debate for the most part. I say it boils down to context, there is no black or white (as with anything).

While it's not necessarily bad practice to test a private method, it's definitely not a requirement. It's also not a requirement to add doc blocks to them either.

Great there's not a rule, so now it boils down to the developer, the situation, and most importantly the code.

If you are building your application correctly, a private method will only manipulate things within itself. They are typically simple little glue methods that take some args and give some output. They typically don't even work with instance variables, but that's not a hard line rule.

By the above definition, there is zero to slim chance that something outside of it can break it. We typically write tests to ward against breaking it outside of when we are working in it, to prevent regressions and the like.

Cool so I basically agree no tests needed. But now the real question again is what does the code do? I have broken some downright simple shit that I would have sworn on my left nut worked. I would be missing multiple left nuts if there was some sort of entity actually taking me up on these swears.

Experience has taught me test every logic, unless you can legit bet your left nut.